### ISSUE TITLE
Switch to semantic versioning

### ISSUE THREAD

**Issue by cboulanger** [qooxdoo/qooxdoo#10109]
Since this is a major break with previous practice, I feel like we should take a formal vote here. 

@johnspackman has already been experimenting with [semantic release](https://github.com/semantic-release/semantic-release), which offers a fully automated release system that is based on formalized commit messages. It will free us from a lot of manual work, but above all, it implies a totally different release versioning model based on [semantic versioning](https://semver.org/), which we already use for our package/library dependency management. 

In particular, it entails that major versions (such as v7, v8) do not mean anything anymore, except that the library has changed in backwards-incompatible ways. While we do not expect to break our library all the time, *any*, even a small non-BC change will increment the major version. This doesn't keep us from releasing a bunch of non-BC changes together in one major version bump. However, we won't have "Big" major version releases ("qooxdoo v7" etc.) anymore. This will allow us to "release early, release often" and concentrate on releasing features, rather than versions. 

This is the model adopted by major players such as Google and Mozilla and I think it is the right one for us, too. With the release of version 6, we should switch to the new model.

**Comment by oetiker** [comment]
Yes, by all means, I am all for going semantic. Since we aim for BC this is great, as it also means that doing something non BC is a 'big' thing and it will motivate us to find ways to avoid this ... :) After all it means for our users that they can use qooxdoo with much more assurance of it being a stable base for their development work.

**Comment by zaucker** [comment]
Make it so!

**Comment by johnspackman** [comment]
it has my vote :)

**Comment by derrell** [comment]
As stated elsewhere, I think there will be some (significant?) hurdles to jump in the process of moving to this new system, but I'm for it. We should learn how to leap those hurdles when we get to them.

**Comment by hkollmann** [comment]
👌 

We need some doc how to create commit messages.

**Comment by johnspackman** [comment]
The commit message format is [the Angular format](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#-git-commit-guidelines)

## Commit Message Format
Each commit message consists of a **header**, a **body** and a **footer**. The header has a special format that includes a **type**, a **scope** and a **subject**:

```
<type>(<scope>): <subject>
<BLANK LINE>
<body>
<BLANK LINE>
<footer>
```

The **header** is mandatory and the **scope** of the header is optional.

Any line of the commit message cannot be longer than 100 characters! This allows the message to be easier to read on GitHub as well as in various git tools.

## Revert
If the commit reverts a previous commit, it should begin with `revert:` , followed by the header of the reverted commit. In the body it should say: `This reverts commit <hash>`., where the hash is the SHA of the commit being reverted.

## Type
Must be one of the following:

- **feat:** A new feature
- **fix:** A bug fix
- **docs:** Documentation only changes
- **style:** Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
- **refactor:** A code change that neither fixes a bug nor adds a feature
- **perf:** A code change that improves performance
- **test:** Adding missing or correcting existing tests
- **chore:** Changes to the build process or auxiliary tools and libraries such as documentation generation


## Scope
The scope could be anything specifying place of the commit change. For example: `compiler`, `framework`, etc...

You can use * when the change affects more than a single scope.

## Subject
The subject contains succinct description of the change:

- use the imperative, present tense: "change" not "changed" nor "changes"
- don't capitalize first letter
- no dot (.) at the end


## Body
Just as in the **subject**, use the imperative, present tense: "change" not "changed" nor "changes". The body should include the motivation for the change and contrast this with previous behavior.

## Footer
The footer should contain any information about **Breaking Changes** and is also the place to reference GitHub issues that this commit closes.

**Breaking Changes** should start with the word `BREAKING CHANGE:` with a space or two newlines. The rest of the commit message is then used for this.


It's basically the docs in https://github.com/semantic-release/semantic-release, and the important part is this table:

Commit message | Release type
-- | --
fix(pencil): stop graphite breaking when too much pressure applied | Patch Release
feat(pencil): add 'graphiteWidth' option | Minor Feature Release
perf(pencil): remove graphiteWidth option
BREAKING CHANGE: The graphiteWidth option has been removed.The default graphite width of 10mm is always used for performance reasons. | Major Breaking Release

**Comment by johnspackman** [comment]
The `scope` is optional, but we could use it to separate out rules for things like PRs - EG if the scope is `compiler` or `docs` then we choose different sets of people to approve (although I dont know how that would be implemented)

**Comment by cboulanger** [comment]
@johnspackman https://github.com/gagoar/use-herald-action seems to do what we want, although one would have to look closer to see if the commit message can be parsed with it to assign reviewers based on the scope.

**Comment by johnspackman** [comment]
perhaps just to ignore the scope altogether, and just use titles like `[compiler]` - that might be more natural, given that we're already used to using things like `[skip ci]` etc etc

The scope probably makes more sense if there are lots of individual modules, but we don't really have those.

One thing that I had assumed (but now wondering whether it's allowed) was that you could have multiple header lines, eg:

```
fix: fixes a bug with doodah (http://github.com/qooxdoo/qooxdoo/issues/1234)
fix: fixes a bug with wotsits (http://github.com/qooxdoo/qooxdoo/issues/1235)
fix: fixes a bug with thingumies (http://github.com/qooxdoo/qooxdoo/issues/1236)
feat: adds new doobry
```

I think that the intention is that one commit === exactly one fix/feat/etc, and as a rule of thumb that's fine but IMHO it should be possible to fix more than one bug report with a single commit, or fix a problem (ie an issue) by adding a feature for example

**Comment by cboulanger** [comment]
I think we can omit the scopes because we modularize on the repository level. https://github.com/gagoar/use-herald-action could assign reviewers based on the commit TYPE (all for a fix:, two for a documentation change,  etc.)

However, once we integrate the compiler, it could make sense to have the optional `(compiler)` scope for a reduced set of reviewers.

**Comment by cboulanger** [comment]
I think this can be closed as we've decided to proceed differently and we've been happy with it so far.

**Comment by hkollmann** [comment]
we did this
